pkg/apis/core/types.go

```diff
@@ -5394,3 +5403,23 @@ type TopologySpreadConstraint struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector
 }
+
+// PortStatus represents the error condition of a service port
+type PortStatus struct {
+	// Port is the port number of the service port of which status is recorded here
+	Port int32
+	// Protocol is the protocol of the service port of which status is recorded here
+	Protocol Protocol
+	// Error is to record the problem with the service port.
```

Comment:
We need to spec the format.  I suggest a condition.type style name (FooBar or company.com/specific-error) where we reserve FooBar style names for a TBD list of conditions.. Maybe we seed that list with "MixedProtocolNotSupported" ?

Reply:
I added a more verbose description and stole the validation of condition.types from metav1. Please check.